### ISSUE TITLE
The way animations are played in Excalibur is non-intuitive

### ISSUE THREAD

**Issue by eonarheim** [excaliburjs/Excalibur#589]
Currently to play an animation you need to call `addDrawing('animationName', anim)` `setDrawing('animationName')` to play an animation, then the animation plays as a side effect.

Problems:
- No way of knowing when the animation is done
- It is ackward to script out an animation sequence
- addDrawing and setDrawing don't scream "use me to animate"
- No way to have an "invisible" animation

Possible implementation

``` typescript

// internal animations collection
var MyActor extends ex.Actor {
   someMethod() {
      // returns a promise that is resolved when the animation is complete
      this.animations('run').play().then(() => {
          this.animations('jump');
      });
   }
}

actor.addAnimation('key', anim);
//alias
actor.add('key', anim)

// in the case of looping animations, the promise resolves when the animation is switched
var done = actor.animations('looping').play()

// the "done" promise resolved
// additionally the "play" animation can take an optional delay parameter
var newanimdon = actor.animations('explode').play(200)

// to stop the current animation, doing a "freeze frame" on the current animation. 
// Resolving the promise of the current animation
actor.animations.stop()

// to have the drawing be invisible call
actor.animations.none();
```

**Comment by kamranayub** [comment]
I like that proposed API

**Comment by alanag13** [comment]
one thing Erik and I discussed about this is that .stop() probably shouldn't resolve the "finished playing"  promise by default. Developers are likely to want different things to happen on a stop(), and since that action is immediate, a promise probably isnt necessary (you should just be able to do whatever you want immediately after calling stop()).

**Comment by eonarheim** [comment]
@alanag13 yes I agree with that.

Additionally, I was thinking I might tweak the proposed `stop()` api and favor `.pause()` and `.resume()`

**Comment by alanag13** [comment]
If at all possible, this should mirror the Sound api, which I noticed has both `.stop()` and `.pause()` but not `.resume()`, which would probably be helpful. In fact, I cannot tell from looking at the sound api if it is even possible to resume a paused sound... in any case I would look carefully at these two APIs and try to have them kinda mesh well.

**Comment by eonarheim** [comment]
Yes, that is probably a good thing to add to this issue. API parity is key.

**Comment by alanag13** [comment]
You could consider doing something like an `IPlayable` interface to keep these sorts of things in check.

**Comment by kamranayub** [comment]
If you have `play` and `resume` I'd make sure to understand the difference. Would `play` restart the sound/animation? I don't think it works that way in most places where this type of control is common (video controls).

I would propose `play()`, `pause()`, and `stop()`. Because when playing, and you pause, pressing play again will resume. If you stop, it resets the "cursor" to the start. This is how most video/music controls work. You'll notice most players will use a single button for play/pause and switch between them based on the state but have a separate Stop button.

![image](https://cloud.githubusercontent.com/assets/563819/14799036/2b9fd486-0b00-11e6-9296-344bc72ae6d5.png)

**Comment by alanag13** [comment]
Agreed; I'm guessing that's the way it must work on `Sound`. The documentation should probably be made a little more clear if this is the case.

**Comment by eonarheim** [comment]
Closed by #1730
